app/api/v1/auth/user.py

```diff
@@ -28,41 +34,75 @@
 router = APIRouter()
 
 
+@router.post(
+    "/find",
+    response_model=UserQueryResponse,
+    dependencies=[ScopedTo([UserScope.admin])],
+    tags=[USER],
+)
+async def find_users(
+    request: Request,
+    skip: int = 0,
+    limit: int = 100,
+    data: UserQueryRequest = Body(...),
+) -> UserQueryResponse:
+    """
+    Find users.
+
+    Search the repository for users that match the filter criteria specified in the request body.
+    If no filter criteria is specified the API will return all users.
+    """
+    filter = write_json_object(data.filter) if data.filter else {}
+    users = filter_user_info(filter, skip, limit, request.app.state.settings)
+    return UserQueryResponse(users=users)
+
+
+scoped_to_any = ScopedTo(
+    [UserScope.admin, UserScope.auditor, UserScope.guardian, UserScope.voter]
+)
+
+
 @router.get(
     "/me",
     response_model=UserInfo,
+    dependencies=[
+        ScopedTo(
+            [UserScope.admin, UserScope.auditor, UserScope.guardian, UserScope.voter]
+        )
+    ],
     tags=[USER],
 )
 async def me(
-    request: Request,
-    scopedTo: ScopedTo = ScopedTo(
-        [UserScope.admin, UserScope.auditor, UserScope.guardian, UserScope.voter]
-    ),
+    request: Request, token_data: ScopedTo = Depends(scoped_to_any)
```

Comment:
Still a still a bit fuzzy on how this all works, but I believe in order to get the current user we need to instantiate a ScopedTo at the file scope and then the `Depends()` call initializes it by calling `__call__` which decodes the JWT token and sets the username in `validate_access_token`.  The previous code that was here wasn't passing the JWT token and so was erroring.  Incidentally I got much of this code from https://fastapi.tiangolo.com/advanced/advanced-dependencies/, but I would be happy if anyone could tell me a better way to do it.

Reply:
I would have referred to the same documentation.